pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -75,6 +75,8 @@ func newReconciler(mgr manager.Manager) (reconcile.Reconciler, metricForwardersM
 type metricForwardersManager interface {
 	Register(types.NamespacedName)
 	Unregister(types.NamespacedName)
+	ProcessError(types.NamespacedName, error)
+	ProcessEvent(types.NamespacedName, datadog.Event)
```

Comment:
maybe you can change the functions signature with the interface that I proposed in previous comment. Like this you dont need anymore the function  `func getNamespacedName(dad *datadoghqv1alpha1.DatadogAgentDeployment) types.NamespacedName ` since from any Object (composition of ObjectMeta) they are all compatible with this interface

```
type metricForwardersManager interface {
	Register(NamespaceName)
	Unregister(NamespaceName)
	ProcessError(NamespaceName, error)
	ProcessEvent(NamespaceName, datadog.Event)
}
```

Reply:
> Like this you don't need anymore the function func getNamespacedName(dad *datadoghqv1alpha1.DatadogAgentDeployment) types.NamespacedName since from any Object (composition of ObjectMeta) they are all compatible with this interface

What if we're processing a `ClusterRole` creation event? we will pass an empty namespace to the forwarder manager and it won't match any metric forwarder and return an error, the event won't be sent

Reply:
by getting the namespacedName from dad we make sure we match the corresponding metric forwarder ID

Reply:
my proposition was to change also the register and unregister function. Like this the forwardID generation is an implementation detail in the forwarder